### PR TITLE
fix(tabs): Add proper RTL support.

### LINKED ISSUE #9292: Scroll support for <md-tabs> in RTL languages
**Actual Behavior**:
Scroll arrows are inverted and scrolling on tabs not functioning
-  `What is the issue? *`
  Scroll support for <md-tabs> in RTL not working
-  `What is the expected behavior?`
  Scroll arrows in correct position and supporting scroll on tabs

**CodePen** (or steps to reproduce the issue): *
Change page direction which includes <md-tab> to rtl (dir="rtl")
-  `CodePen Demo which shows your issue:`
  http://codepen.io/anon/pen/oLJKNr
-  `Details:`
  open up developer tools/ Inspect element (F12) to narrow screen size.
  Adjust until the tabs are truncated and the next/previous arrows appear
  Notice the arrows are inverted and the scroll left/right no longer works
  Is there a fix for this issue?

**Angular Versions**: *
-  `Angular Version:`
  v1.5.0
-  `Angular Material Version:`
  v1.1.0-rc4

**Additional Information**:
-  `Browser Type: *` Chrome
-  `Browser Version: *`52.0.2743.82 m
-  `OS: *` Windows
-  `Stack Traces:` ---

---

Shortcut to create a [new CodePen Demo](http://codepen.io/team/AngularMaterial/pen/bEGJdd).
Note: `*` indicates required information. Without this information, your issue may be auto-closed.

> Do not modify the titles or questions. Simply add your responses to the ends of the questions.
>   Add more lines if needed.

### PR DESCRIPTION
**Pull request by topherfangio**
Tabs currently attempts no RTL support for scrolling and pagination.

Update component to properly support RTL.

 - Add new MdTabsPaginationService to handle some calculations in
   a more testable way.
 - Add many instances of RTL checks/math.
 - Add minimum tab width based on spec (or tabs width if smaller).
 - Update SCSS to account for RTL mode with pagination icons.
 - Also add check to ensure pagination buttons ignore clicks when
   disabled.

Fixes #9292.

### CODE REVIEW

File: src/components/tabs/js/tabsController.js [hunk 1]
```diff
@@ -325,28 +328,61 @@ function MdTabsController ($scope, $element, $window, $mdConstant, $mdTabInkRipp
...
+
     var elements = getElements();
     var viewportWidth = elements.canvas.clientWidth,
         totalWidth    = viewportWidth + ctrl.offsetLeft,
         i, tab;
-    for (i = 0; i < elements.tabs.length; i++) {
-      tab = elements.tabs[ i ];
-      if (tab.offsetLeft + tab.offsetWidth > totalWidth) break;
+
+    if (isRtl()) {
```
Comment (EladBezalel) [inline src/components/tabs/js/tabsController.js]: Looks like the logic in the if and else are very similar

**Comment by EladBezalel** [comment]
This is big! For now lgtm, though I want to manual test it

**Comment by EladBezalel** [comment]
This looks LGTM!!
Really want this one asap :pray:

**Comment by NaNgets** [comment]
This is really urgent for me,
any chance to merge it soon?

File: src/components/tabs/js/tabsController.js (hunk 1 above)
Comment (ThomasBurleson) [inline src/components/tabs/js/tabsController.js]: Please refactor!

**Comment by topherfangio** [comment]
@NaNgets As I worked with it a bit more and rebased it against the latest changes, I realized it had some problems and I've started refactoring all of the math behind the pagination into a separate service that I can more easily test. This work is under way, but I've had some other high-priority issues come up that need to be fixed first.

I'm aiming to get this in 1.1.2 though.

**Comment by NaNgets** [comment]
Thank you very much! :)
Looking forward to see it and have it merged.

**Comment by gkorland** [comment]
When is the 1.1.2 expected to be released?

**Comment by mrhammadasif** [comment]
+1

**Comment by ThomasBurleson** [comment]
@topherfangio - can you finish the refactors on this for 1.1.4 release ?

**Comment by ThomasBurleson** [comment]
@topherfangio - plz rebase.

**Comment by amitmiran** [comment]
+1 
This is urgent for me 
what is the 1.1.5  release date?

**Comment by Chukatuk** [comment]
+1
Jumping in on the "this is urgent for me too" train =))

**Comment by Chukatuk** [comment]
For the super impatient (like me):
I merged the commit with material 1.1.4 and registered it as bower package - "angular-material-rtl-tabs".
Everything seems to be working fine until now :)

**Comment by jelbourn** [comment]
@tinayuangao can you run a presubmit on this for `latest/` and `v1_1`

**Pull request merged**